src/traverse/Indicator.ts

```diff
@@ -34,7 +34,11 @@ export class ContractDefinitionIndicator {
   }
 
   updateIncrementation(path: NodePath, state: any) {
-    if (!path.isIncremented || state.incrementedIdentifier.isKnown) {
+    // if an incrementation is marked as unknown anywhere, the binding will know
+    if (
+      !path.isIncremented ||
+      state.incrementedIdentifier.isKnown || !path.scope.getReferencedBinding(state.incrementedIdentifier).isUnknown
```

Comment:
added a check that the incremented state isn't marked as unknown anywhere. Since now we assume it's whole, this change may cause issues, but I haven't found any yet

Reply:
TO EDIT